test/krew_test.go

```diff
@@ -36,13 +58,117 @@ func TestKrewInstall(t *testing.T) {
 	test.Call(validPlugin, "--help").RunOrFail()
 }
 
-func TestKrewHelp(t *testing.T) {
+func TestKrewUninstall(t *testing.T) {
 	skipShort(t)
 
 	test, cleanup := krew.NewTest(t)
 	defer cleanup()
 
-	test.Krew("help").RunOrFail()
+	test.WithIndex().Krew("install", validPlugin).RunOrFailOutput()
+	test.Krew("uninstall", validPlugin).RunOrFailOutput()
+	if err := test.Call(validPlugin, "--help").Run(); err == nil {
+		t.Errorf("Expected the plugin to be uninstalled")
+	}
+}
+
+func TestKrewSearchAll(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := krew.NewTest(t)
+	defer cleanup()
+
+	output := test.WithIndex().Krew("search").RunOrFailOutput()
+	if plugins := lines(output); len(plugins) < 10 {
+		// the first line is the header
+		t.Errorf("Expected at least %d plugins", len(plugins)-1)
+	}
+}
+
+func TestKrewSearchOne(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := krew.NewTest(t)
+	defer cleanup()
+
+	plugins := lines(test.WithIndex().Krew("search", "krew").RunOrFailOutput())
+	if len(plugins) < 2 {
+		t.Errorf("Expected krew to be a valid plugin")
+	}
+	if !strings.HasPrefix(plugins[1], "krew ") {
+		t.Errorf("The first match should be krew")
+	}
+}
+
+func TestKrewInfo(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := krew.NewTest(t)
+	defer cleanup()
+
+	output := test.WithIndex().Krew("info", validPlugin).RunOrFailOutput()
```

Comment:
again, here we probably just need exitcode=0 (+ I realize we don't have tests for true negatives such as "info for nonexisting plugin fails"; we can do those later) as that's the behavior that this cmd offers (the NAME: foo) isn't part of a supported API.

Reply:
Yeah right. For good measure, I already added the negative test case.